Dockerfile

```diff
@@ -161,13 +161,16 @@ ENV PATH="/root/stp/build:${PATH}"
 ENV LD_LIBRARY_PATH="/root/stp/deps/cadical/build:/root/stp/deps/cadiback/:$LD_LIBRARY_PATH"
 
 # Build Yosys.
-WORKDIR /root
-RUN source /root/dependencies.sh \
-  && mkdir yosys && cd yosys \
-  && wget -qO- https://github.com/YosysHQ/yosys/archive/$YOSYS_COMMIT_HASH.tar.gz | tar xz --strip-components=1 \
-  && PREFIX="/root/.local" CPLUS_INCLUDE_PATH="/usr/include/tcl8.6/:$CPLUS_INCLUDE_PATH" make config-gcc \
-  && PREFIX="/root/.local" CPLUS_INCLUDE_PATH="/usr/include/tcl8.6/:$CPLUS_INCLUDE_PATH" make -j ${MAKE_JOBS} install \
-  && rm -rf /root/yosys
+WORKDIR /root 
+RUN cd /root \
+&& git clone https://github.com/YosysHQ/yosys.git \
```

Comment:
I would prefer you use the format that I'm already using, i.e.
```
  && wget -qO- https://github.com/YosysHQ/yosys/archive/$YOSYS_COMMIT_HASH.tar.gz | tar xz --strip-components=1 \
```

you should be able to just keep that line as-is! Is there a reason you had to switch to checking out the repo instead of downloading an archived version?

In fact, why did this part of the Dockerfile have to change at all?

Reply:
So that's a good question. Basically, I couldn't find an archived version that would work with that line. 

The motivation behind updating the yosys version is that the previous yosys version can't handle certain SystemVerilog syntax IIRC. Basically, the whole `bsg_mul_add_unsigned` PR had to be modified partially because the older yosys version couldn't handle some of the SV syntax present. 

By updating the version of yosys, newer PRs wouldn't have to be modified (at least, they'd have to be modified less)

Reply:
Do you know how we could get the most recent yosys version to work with this line? 
`  && wget -qO- https://github.com/YosysHQ/yosys/archive/$YOSYS_COMMIT_HASH.tar.gz | tar xz --strip-components=1 \` 

Whenever I would try to use the most recent Yosys commit, it would fail CI

Reply:
So it looks like what happened is this: Yosys added their fork of `abc` as a submodule. Using the archive method of downloading a repo does not download submodules (see https://github.com/dear-github/dear-github/issues/214). So we do in fact have to go back to checking it out using git.

Just to be clear: in general, I really like to avoid assuming things are a git repository during build processes. I find it to be an anti-pattern. Avoiding the use of git here was just trying to avoid the anti-pattern, but it's actually not a huge deal.

I'm going to leave one more comment about what to do about this! But otherwise this change is OK.